core/templates/components/button-directives/create-activity-button.component.spec.ts

```diff
@@ -150,44 +150,127 @@ describe('CreateActivityButtonComponent', () => {
+      expect(component.initCreationProcess).toHaveBeenCalled();
+    }));
+
+    it('should redirect user to creator dashboard page when the user' +
+      ' clicks on the create button', () => {
+      component.creationInProgress = false;
+      component.canCreateCollections = true;
+      spyOn(urlService, 'getPathname').and.returnValue('not/creator-dashboard');
+      spyOnProperty(windowRef, 'nativeWindow').and.returnValue({
+        location: {
+          replace: (val: string) => {}
+        }
+      });
+      const modalSpy = spyOn(ngbModal, 'open').and.callFake((dlg, opt) => {
+        return <NgbModalRef>({
+          result: Promise.resolve('success')
+        });
+      });
+      const replaceSpy = spyOn(windowRef.nativeWindow.location, 'replace');
 
-    spyOn(component, 'initCreationProcess');
+      component.initCreationProcess();
 
-    expect(component.userIsLoggedIn).toBe(false);
-    expect(component.canCreateCollections).toBe(false);
+      expect(replaceSpy).toHaveBeenCalledWith('/creator-dashboard?mode=create');
+      expect(modalSpy).not.toHaveBeenCalled();
+    });
 
-    component.ngOnInit();
-    tick();
-    fixture.detectChanges();
+    it('should not redirect user but open a create activity modal if user' +
+      ' is on the creator dashboard page', () => {
+      component.creationInProgress = false;
+      component.canCreateCollections = true;
+      spyOn(urlService, 'getPathname').and.returnValue(
+        '/creator-dashboard');
+      const modalSpy = spyOn(ngbModal, 'open').and.callFake((dlg, opt) => {
+        return <NgbModalRef>({
+          result: Promise.resolve('success')
+        });
+      });
+      const replaceSpy = spyOn(windowRef.nativeWindow.location, 'replace');
 
-    expect(component.userIsLoggedIn).toBe(true);
-    expect(component.canCreateCollections).toBe(true);
-    expect(component.initCreationProcess).toHaveBeenCalled();
-  }));
+      component.initCreationProcess();
 
-  it('should create a new exploration automatically if the user cannot create' +
-    ' collections and is redirected to creator dashboard after' +
-    ' logging in', fakeAsync(() => {
-    spyOn(userService, 'getUserInfoAsync')
-      .and.returnValue(Promise.resolve(userInfoForNonCollectionCreator));
-    const explorationCreationServiceSpy = spyOn(
-      explorationCreationService, 'createNewExploration');
+      expect(replaceSpy).not.toHaveBeenCalled();
+      expect(modalSpy).toHaveBeenCalled();
+    });
 
-    expect(component.userIsLoggedIn).toBe(false);
-    expect(component.canCreateCollections).toBe(false);
+    it('should handle modal\'s failure callback' +
+    ' and stop creation process', fakeAsync(() => {
+      component.creationInProgress = false;
+      component.canCreateCollections = true;
+      spyOn(urlService, 'getPathname').and.returnValue(
+        '/creator-dashboard');
+      const modalSpy = spyOn(ngbModal, 'open').and.callFake((dlg, opt) => {
```

Comment:
Can this spy on a specific modal (or be applied to a specific modal) rather than all modals in general?

Reply:
modalSpy can only spy on all modal in general, but we can check if it was called with a specific spy. Changed the test